src/main/scala/com/amazon/deequ/analyzers/KLLSketch.scala

```diff
@@ -0,0 +1,122 @@
+package com.amazon.deequ.analyzers
+
+import com.amazon.deequ.analyzers.catalyst.KLLSketchSerializer
+import com.amazon.deequ.analyzers.runners.IllegalAnalyzerParameterException
+import com.amazon.deequ.metrics.{BucketDistribution, BucketValue, KLLMetric}
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+import java.nio.ByteBuffer
+import com.amazon.deequ.analyzers.Analyzers._
+import com.amazon.deequ.analyzers.Preconditions.{hasColumn, isNumeric}
+import com.amazon.deequ.analyzers.runners.MetricCalculationException
+import org.apache.spark.sql.DeequFunctions.stateful_kll
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{Column, Row}
+
+import scala.util.Failure
+
+/** State for KLL Sketches */
+case class KLLState(qSketch: QuantileNonSample[Double], globalMax: Double, globalMin: Double)
+  extends State[KLLState] {
+
+  /** Add up states by merging sketches */
+  override def sum(other: KLLState): KLLState = {
+    val mergedSketch = qSketch.merge(other.qSketch)
+    KLLState(mergedSketch, Math.max(globalMax,other.globalMax), Math.min(globalMin,other.globalMin))
+  }
+}
+
+object KLLState{
+  // from Array[Byte] to original type
+  def fromBytes(bytes: Array[Byte]): KLLState = {
+    val buffer = ByteBuffer.wrap(bytes)
+    val min = buffer.getDouble
+    val max = buffer.getDouble
+    val store = new Array[Byte](buffer.remaining())
+    buffer.get(store)
+    val obj = KLLSketchSerializer.serializer.deserialize(store)
+    KLLState(obj, max, min)
+  }
+
+}
+
+/**
+ *
+ *
+ * @param column Which column to compute this aggregation on.
+ */
+case class KLLSketch(column: String, where: Option[String] = None, sketchSize: Int = KLLSketch.defaultSketchSize,
```

Comment:
Please format according to our [coding conventions](https://github.com/databricks/scala-style-guide)

Reply:
Done.